woocommerce-gutenberg-products-block.php

```diff
@@ -153,90 +153,49 @@ function() {
-	// JSON files with the file they should be included for. This is an md5
-	// of 'packages/woocommerce-blocks/build/FILENAME.js'.
-	$core_path_md5     = md5( 'packages/woocommerce-blocks/build/' . $handle_filename );
-	$core_json_file    = $lang_dir . '/woocommerce-' . $locale . '-' . $core_path_md5 . '.json';
-	$json_translations = is_file( $core_json_file ) && is_readable( $core_json_file ) ? file_get_contents( $core_json_file ) : false; // phpcs:ignore
-
-	if ( ! $json_translations ) {
-		return $translations;
-	}
-
-	// Rather than short circuit pre_load_script_translations, we will output
-	// core translations using an inline script. This will allow us to continue
-	// to load feature-plugin translations which may exist as well.
-	$output = <<<JS
-	( function( domain, translations ) {
-		var localeData = translations.locale_data[ domain ] || translations.locale_data.messages;
-		localeData[""].domain = domain;
-		wp.i18n.setLocaleData( localeData, domain );
-	} )( "{$domain}", {$json_translations} );
-JS;
-
-	if ( empty( $wp_scripts->done ) ) {
-		// If we hadn't printed any script into the page, let's enqueue the translations.
-		// phpcs:ignore WordPress.WP.EnqueuedResourceParameters.NoExplicitVersion
-		wp_register_script( $handle_filename, '', array( 'wp-i18n' ), false, true );
-		wp_enqueue_script( $handle_filename );
-		wp_add_inline_script(
-			$handle_filename,
-			$output
-		);
-	} else {
-		// If we have already printed scripts into the page, there is a chance that
-		// scripts have finished being printed. That means that if we enqueued them here,
-		// they would never be printed. Instead of enqueuing, then, let's print directly
-		// the script tag.
-		printf( "<script type='text/javascript'>\n%s\n</script>\n", $output ); // phpcs:ignore
+	/**
+	 * We only care about the translation file of the feature plugin in the
+	 * wp-content/languages folder.
+	 */
+	if ( false === strpos( $file, $lang_dir ) ) {
+		return $file;
 	}
 
-	// Finally, short circuit the pre_load_script_translations hook by returning
-	// the translation JSON from the feature plugin, if it exists so this hook
-	// does not run again for the current handle.
-	$path_md5     = md5( 'build/' . $handle_filename );
-	$json_file    = $lang_dir . '/' . $domain . '-' . $locale . '-' . $path_md5 . '.json';
-	$translations = is_file( $json_file ) && is_readable( $json_file ) ? file_get_contents( $json_file ) : false; // phpcs:ignore
-
-	if ( $translations ) {
-		return $translations;
+	// If the translation file for feature plugin exist, use it.
+	if ( is_readable( $file ) ) {
+		return $file;
 	}
 
-	// Return valid empty Jed locale.
-	return '{ "locale_data": { "messages": { "": {} } } }';
+	/**
+	 * Return file path of the corresponding translation file in the WC Core
+	 * here is enough because `load_script_translations()` will check for its
+	 * existence before loading it.
+	 */
+	return str_replace( $file, "/plugins/{$domain}-", '/plugins/woocommerce-' );
```

Comment:
I might be misunderstanding this, but if `$file` is the subject, shouldn't it be the [third parameter](https://www.php.net/manual/en/function.str-replace.php#refsect1-function.str-replace-parameters)?

Besides that, I think the md5 hash will be different between WC core and WC Blocks, so that's something we will need to replace too when trying to fetch the translations from core, no?

Reply:
> I might be misunderstanding this, but if $file is the subject, shouldn't it be the [third parameter](https://www.php.net/manual/en/function.str-replace.php#refsect1-function.str-replace-parameters)?

It's definitely my wrong function usage 😅

Reply:
> Besides that, I think the md5 hash will be different between WC core and WC Blocks, so that's something we will need to replace too when trying to fetch the translations from core, no?

You're right! I reused the existing code to fix the issue you mentioned here.